taiga_halo2/examples/sudoku/app/dealer_intent_vp.rs

```diff
@@ -0,0 +1,417 @@
+use ff::Field;
+use halo2_gadgets::{
+    poseidon::{
+        primitives as poseidon, primitives::ConstantLength, Hash as PoseidonHash,
+        Pow5Chip as PoseidonChip,
+    },
+    utilities::bool_check,
+};
+use halo2_proofs::{
+    circuit::{floor_planner, AssignedCell, Layouter, Region, Value},
+    plonk::{
+        keygen_pk, keygen_vk, Advice, Circuit, Column, ConstraintSystem, Constraints, Error,
+        Instance, Selector,
+    },
+    poly::Rotation,
+};
+use pasta_curves::pallas;
+use rand::rngs::OsRng;
+use rand::RngCore;
+use taiga_halo2::{
+    circuit::{
+        gadgets::{assign_free_advice, GetTargetNoteVariableConfig},
+        integrity::{OutputNoteVar, SpendNoteVar},
+        note_circuit::NoteConfig,
+        vp_circuit::{
+            VPVerifyingInfo, ValidityPredicateCircuit, ValidityPredicateConfig,
+            ValidityPredicateInfo,
+        },
+    },
+    constant::{NUM_NOTE, SETUP_PARAMS_MAP},
+    note::Note,
+    proof::Proof,
+    utils::poseidon_hash,
+    vp_circuit_impl,
+    vp_vk::ValidityPredicateVerifyingKey,
+};
+
+#[derive(Clone, Debug, Default)]
+struct DealerIntentValidityPredicateCircuit {
+    spend_notes: [Note; NUM_NOTE],
+    output_notes: [Note; NUM_NOTE],
+    // The note that vp owned is set at spend_notes[0] or output_notes[0] by default. Make it mandatory later.
+    // is_spend_note helps locate the target note in spend_notes and output_notes.
```

Comment:
Do you think it's worth sketching how locating the target note happens here in plain text?